Showcase 32 fix scenarios screen

ronal2do opened the pull request:
Refactor Types and Scenarios with Improved Type Safety
- Introduced `scenarioToFormData` transformer for consistent data conversion
- Improved type safety in scenario editing and step management
- Extracted common utility functions like `convertBase64`
- Updated Zod schemas to use literal types and more precise validations
- Enhanced test coverage for scenario-related functionality

Replace FontAwesome icons with Lucide icons
- Removed @fortawesome/react-fontawesome and @fortawesome/free-solid-svg-icons dependencies
- Replaced FontAwesomeIcon components with Lucide icons across multiple components

Update package dependencies and lock file

- Upgrade cmdk from 1.0.0 to 1.0.4
- Add @dnd-kit/react package
- Update tslib to version 2.8.1
- Remove redundant type dependencies

ronal2do committed:
Refactor Types and Scenarios with Improved Type Safety
- Introduced `scenarioToFormData` transformer for consistent data conversion
- Improved type safety in scenario editing and step management
- Extracted common utility functions like `convertBase64`
- Updated Zod schemas to use literal types and more precise validations
- Enhanced test coverage for scenario-related functionality

ronal2do committed:
Replace FontAwesome icons with Lucide icons
- Removed @fortawesome/react-fontawesome and @fortawesome/free-solid-svg-icons dependencies
- Replaced FontAwesomeIcon components with Lucide icons across multiple components

ronal2do committed:
Update package dependencies and lock file

- Upgrade cmdk from 1.0.0 to 1.0.4
- Add @dnd-kit/react package
- Update tslib to version 2.8.1
- Remove redundant type dependencies

ronal2do committed:
Merge branch 'main' into SHOWCASE-32-fix-scenarios-screen

ronal2do committed:
Add unit tests for utility functions
- Create test suite for `cn` class name merging utility
- Add test for `convertBase64` file conversion utility

ronal2do committed:
Enhance GitHub Actions workflow with comprehensive CI checks

- Add test, lint, and typecheck steps to CI pipeline
- Ensure build process runs after all checks pass

ronal2do committed:
Update ESLint configuration and Next.js ESLint package

- Disable TypeScript version warning in ESLint configuration
- Upgrade eslint-config-next to version 15.1.6

BtencateSphereon commented on components/scenario-screen/edit-step-screen.tsx:
```diff
@@ -9,83 +9,52 @@ import { NoSelection } from "@/components/credentials/no-selection";
-      {stepState === "adding-step" && (
-        <ChooseStepType addNewStep={handleAddStep} />
+      {(stepState === "none-selected" || stepState === null) && (
+        <NoSelection text={"Nothing Selected"} />
```
should be getting a localization string as this is now available

ronal2do replied:
ok

BtencateSphereon commented on components/scenario-screen/display-step-credentials.tsx:
```diff
@@ -3,22 +3,24 @@ import { Trash2, Edit } from "lucide-react";
   };
-  selectedStep: number;
-  selectedScenario: number;
+  selectedStep: number | null;
```
just checking, any specific reason why we are using null here?

ronal2do replied:
We can have no selected steps in order to display`<NoSelection`

BtencateSphereon commented on components/scenario-screen/local-file-upload.tsx:
```diff
@@ -20,22 +20,10 @@ export function LocalFileUpload({
   useEffect(() => {
-    setPreview(localJSON[element] || null);
+    const [section, field] = element;
+    setPreview(localJSON[section]?.[field] || null);
```
should the nullish coalescing be used here?

BtencateSphereon commented on components/scenario-screen/proof-attribute.tsx:
```diff
@@ -29,23 +35,27 @@ export const ProofAttribute = ({
-            </Select.Option>
-          ))}
+          <SelectTrigger>
+            <SelectValue placeholder="Select attribute" />
```
should be localized?

BtencateSphereon commented on components/scenario-screen/scenario-edit.tsx:
```diff
@@ -26,38 +28,31 @@ export const ScenarioEdit = () => {
-      steps: currentScenario?.steps || [], // Preserve existing steps
+      overview: {
+        ...data.overview,
+        image: data.overview.image || '',
```
nullish coalescing? is see this on some places where i think this would be better suited than the logical or

BtencateSphereon commented on components/scenario-screen/scenario-edit.tsx:
```diff
@@ -99,18 +93,23 @@ export const ScenarioEdit = () => {
-                  shouldTouch: true,
-                  shouldValidate: true,
-                })
+              element={["overview", "image"] as [string, string]}
```
should we not make some sort of string literal type for this?

BtencateSphereon commented on components/scenario-screen/scenario-screen.tsx:
```diff
@@ -127,7 +130,7 @@ export const ScenarioScreen = () => {
               </DndContext>
 
-              <Button
+              {/* <Button
```
remove commented code?

BtencateSphereon commented on components/scenario-screen/scenario-step.tsx:
```diff
@@ -52,93 +52,80 @@ export const ScenarioStep = ({
+              <hr />
+              <div className="w-full py-2">
+                <p className="text-sm m-1 mt-2 font-bold">
+                  Requested Credentials:
```
localization missing

BtencateSphereon commented on components/scenario-screen/scenario-step.tsx:
```diff
@@ -52,93 +52,80 @@ export const ScenarioStep = ({
+              {step.text.length > MAX_CHARS ? (
+                <>
+                  {step.text.slice(0, MAX_CHARS)}...{" "}
+                  <span className="font-bold">see more</span>
```
localization missing

BtencateSphereon commented on hooks/use-scenarios.ts:
```diff
@@ -192,15 +211,16 @@ export const useScenarios = create<State & Actions>((set, get) => ({
-  type: isProof ? "proof" : "basic",
-  title: "",
+  type: type,
+  title: type === StepType.CONNECT_AND_VERIFY ? "Confirm the information to send" : "",
```
localization

BtencateSphereon reviewed (changes requested):
just some small notes / questions

ronal2do commented:
@BtencateSphereon I'll sync with your branch to add the localization

ronal2do committed:
Merge branch 'main' into SHOWCASE-32-fix-scenarios-screen

ronal2do committed:
- Update LocalFileUpload component with improved type handling
- Add translation support to ProofAttribute and ScenarioStep components
- Modify scenario step creation with more flexible title initialization
- Update localization files with new translation keys
- Remove commented-out code and improve null coalescing

ronal2do committed:
Merge branch 'intl' into SHOWCASE-32-fix-scenarios-screen

ronal2do committed:
refactor: improve navbar layout and styling with responsive grid design

ronal2do committed:
feat: add skeleton loading state for dark mode toggle

ronal2do committed:
Merge pull request #10 from Sphereon-Opensource/SHOWCASE-20-github-actions-for-build

Showcase 20 GitHub actions for build

ronal2do committed:
Merge branch 'main' into SHOWCASE-32-fix-scenarios-screen

ronal2do committed:
refactor: remove react-i18next and update translation imports

ronal2do committed:
chore: add eslint-plugin-react-hooks as a dev dependency
